packages/terra-paginator/src/ControlledPaginator.jsx

```diff
@@ -114,10 +113,6 @@ class Paginator extends React.Component {
     return pageButtons;
   }
 
-  hasNavContext() {
-    return this.props.totalCount && this.props.itemCountPerPage;
-  }
```

Comment:
This seems like it played an important role in the logic statements below, can you explain why the check is not necessary? `totalCount` and `itemCountPerPage` are marked as required props.

Reply:
This check was removed to ensure paginator to consider `totalCount` when `itemCountPerPage` was not provided. I missed to notice that `itemCountPerPage` was required prop in controlled-paginator. 

This is bit confusing so with existing design only paginator supports unknown Page count...? and controlled-paginator is not supposed to support unknown Page counts. If yes then I need to add this check back OR if controlled-paginator should also updated to support unknown page counts then we need to remove `required` from props `totalCount` and `itemCountPerPage` .

Reply:
Spoke with @StephenEsser, @neilpfeiffer, and others and we should support unknown page counts and we should go ahead and make totalCount and itemCountPerPage optional.